dist/jquery.validate.unobtrusive.js

```diff
@@ -419,6 +419,9 @@
             setValidationValues(options, "regex", options.params.regex);
         }
     });
+    adapters.add("fileextensions", ["extensions"], function (options) {
```

Comment:
Just checking - do jQuery and Data Annotations use the exact same format for the list of extensions?

Do they support:
* `*.foo, *.bar` (comma-separated, with leading `*.`)
* `.foo, .bar` (comma-separated, leading `.`)
* `.foo; .bar` (semicolon-separated, with leading `.`)

Etc.?

Reply:
From here: https://jqueryvalidation.org/extension-method/
Here a leading '.' or '*.' is not a valid string and also data annotations extensions are expected without the these too
```
extension( [extension ] )
extension
Type: String
The allowed file extensions, seperated with "|" (or a comma, ","), defaults to "png|jpe?g|gif".
```

Reply:
@kichalla the .NET Core `FileExtensionsAttribute` does not support wildcards nor separator characters other than `','`. Sounds like the jQuery method isn't a good match.

Reply:
Hmm, `FileExtensionsAttribute` ignores all spaces and periods in the `Extensions` property. Then it adds a leading period before doing the actual matching. See [here](https://github.com/dotnet/corefx/blob/15f9a31fd45f9a7d667765ef08ecfd1cefddde3a/src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/FileExtensionsAttribute.cs#L35-L41).

If we use the `extension` function and it doesn't do both of the above, we need additional special cases in the C# code or something more complicated here.

Reply:
We could presumably massage the attribute's property into the format that jQuery Validation expects, no? It seems fairly trivial to do so. Just do the same stripping of unnecessary characters (`.` and ` `), and that's it, right? (I'm not worried about handling crazy stuff like if someone passed in `foo*.jpg, foo...bar...baz` because that's nonsensical to begin with.)

Reply:
I created the following PR now: https://github.com/aspnet/Mvc/pull/6860